greenpithumb/dht11.py

```diff
@@ -0,0 +1,50 @@
+import datetime
+import threading
+
+from Adafruit_DHT import common
+
+_FRESHNESS_THRESHOLD = 2
+
+
+class CachingDHT11(object):
+    """Caching DHT11 sensor wrapper.
+
+    Reads and returns temperature and humidity levels while also caching these
+    values to ensure that the sensor is not polled at too high of a frequency.
+    """
+
+    def __init__(self, dht11_pin, clock):
+        """Creates a new CachingDHT11 object.
+
+        Args:
+            dht11_pin: An int representing the pin that the DHT11 sensor is
+                connected to.
+            clock: A clock interface
+        """
+        self._dht11_pin = dht11_pin
+        self._clock = clock
+        self._last_reading_time = datetime.min
+        self._last_reading = None
+        self._lock = threading.Lock()
+
+    def humidity(self):
+        """Returns a recent relative humidity reading."""
+        self._lock.acquire()
+        now = self._clock.now()
+        if (now - self._last_reading_time).total_seconds >= (
+                _FRESHNESS_THRESHOLD):
+            self._last_reading_time = now
+            self._last_reading = common.read(11, self._dht11_pin)
+        self._lock.release()
+        return self._last_reading[0]
```

Comment:
Suggest replacing the `0` here and `1` below with named constants so we avoid magic numbers.